### PR TITLE
Feat: Add `approval` and `approval approver` to openapi spec and get-job-status action

### PR DESCRIPTION
**Pull request by zacharyblasczyk**
## Release Notes

- **New Features**
  - Enhanced job detail retrieval now includes approval status and approver information.
  - Introduced new approval-related data structures in the API response for improved context on job approvals.

- **Bug Fixes**
  - Improved error handling during API calls and output settings.

- **Documentation**
  - Updated OpenAPI specifications to reflect new approval properties and response structures.

- **Chores**
  - Updated configuration files to align with the new OpenAPI specification.

### CODE REVIEW

File: openapi.v1.yaml [hunk 1]
```diff
@@ -391,15 +386,57 @@ paths:
...
+                            type: string
+                          email:
+                            type: string
+                        required:
+                          - id
+                          - name
+                          - email
+                    required:
+                      - id
+                      - status
```
[the remark this answers is not in the document]
Comment (zacharyblasczyk) [reply openapi.v1.yaml:431]: @jsbroks, I had the exact same idea about adding a statusUpdatedAt to the table so we could show in the UI "approved/rejected at Date()" I just wanted to do that in a followup PR.

**Comment by zacharyblasczyk** [comment]
Still testing this.

File: apps/webservice/src/app/api/v1/jobs/[jobId]/route.ts [hunk 2]
```diff
@@ -49,8 +52,48 @@
       target: row.target,
       deployment: row.deployment,
       release: row.release,
+      causedBy: row.user
+        ? {
+            id: row.user.id,
+            name: row.user.name,
+            email: row.user.email,
+          }
+        : null,
```
[the remark this answers is not in the document]
Comment (jsbroks) [reply apps/webservice/src/app/api/v1/jobs/[jobId]/route.ts:61]: @zacharyblasczyk agree on this, remove email

File: integrations/github-get-job-inputs/src/index.ts [hunk 3]
```diff
@@ -75,11 +83,20 @@ async function run() {
       setOutputsRecursively("release_config", release?.config);
       setOutputsRecursively("release_metadata", release?.metadata);
 
+      setOutputAndLog("caused_by_id", causedBy.id);
+      setOutputAndLog("caused_by_name", causedBy.name);
+      setOutputAndLog("caused_by_email", causedBy.email);
```
Comment (jsbroks) [inline integrations/github-get-job-inputs/src/index.ts:88]: we shouldn't need this

**Review by jsbroks — APPROVED** [review]

**Pull request merged**